### PR TITLE
update dev iteration docs 

### PR DESCRIPTION
**Pull request by jrhizor**
Make the commands to run simpler/more consistent in the documentation. I'm not sure why so many unrelated changes are showing up when I create a variant from the main documentation. I might just update this without Gitbook.

### CODE REVIEW

File: docs/contributing/contributing-to-airbyte/developing-locally.md [hunk 1]
```diff
@@ -29,8 +29,8 @@ This will build all the code and run all the unit tests.
 ## Run in `dev` mode with `docker-compose`
 
 ```bash
-./gradlew build
-docker-compose --env-file .env.dev -f docker-compose.yaml -f docker-compose.dev.yaml up
+./gradlew build composeBuild
```
Comment (jrhizor) [inline docs/contributing/contributing-to-airbyte/developing-locally.md:32]: this file has the only relevant changes

**Review by sherifnada — APPROVED** [review]

**Comment by michel-tricot** [comment]
Gitbook has problem. I am trying to get access to their support but no answer so far

File: docs/contributing/contributing-to-airbyte/developing-locally.md (hunk 1 above)
Comment (michel-tricot) [reply docs/contributing/contributing-to-airbyte/developing-locally.md:32]: composeBuild is included in build already

File: docs/contributing/contributing-to-airbyte/developing-locally.md [hunk 2]
```diff
@@ -29,8 +29,8 @@ This will build all the code and run all the unit tests.
...
+VERSION=dev docker-compose up
```
Comment (michel-tricot) [inline docs/contributing/contributing-to-airbyte/developing-locally.md:33]: This is incorrect.

The title says:
"Run in `dev` mode with `docker-compose`" not run with the dev images.

It sets all the options so you can develop locally and use the local file system

**Review by michel-tricot — CHANGES_REQUESTED** [review]
What was motivating that change? The previous instructions were correct.

File: docs/contributing/contributing-to-airbyte/developing-locally.md (hunk 2 above)
Comment (michel-tricot) [reply docs/contributing/contributing-to-airbyte/developing-locally.md:33]: the goal of that command is not to run it but to run it so you can develop

**Comment by jrhizor** [comment]
I think it's clear I didn't understand the purpose of these commands completely. I'll nix these changes and just add clarification to what the sections mean.

**Pull request closed without merging**